cmake/GtsamMakeConfigFile.cmake

```diff
@@ -20,13 +20,28 @@ function(GtsamMakeConfigFile PACKAGE_NAME)
 		set(EXTRA_FILE "_does_not_exist_")
 	endif()
 
+	# Version file
+	include(CMakePackageConfigHelpers)
+	write_basic_package_version_file(
+	  "${PROJECT_BINARY_DIR}/${PACKAGE_NAME}ConfigVersion.cmake"
+	  VERSION ${GTSAM_VERSION_STRING}
+	  COMPATIBILITY AnyNewerVersion
```

Comment:
I would change compatibility to either `SameMajorVersion` or `SameMajorVersion`. I vote for the former to make people's live easier. This is because, the way we do semantic versioning, we don't guarantee backwards compatibility between major version changes. So if and when GTSAM-5 comes out, Cmake won't complain, but the compiler will, and that can be confusing.

@dellaert your thoughts?

Reply:
No problem changing to `SameMajorVersion`, I also think it makes more sense.

Reply:
Changed compatibility to `SameMajorVersion`

Reply:
Awesome!